_posts/function-contracts.md

```diff
@@ -0,0 +1,298 @@
+# Function Contracts for Kani
+
+I this blogpost we discuss a new feature we’re developing for Kani: Function Contracts. It’s now available as an unstable feature with the `-Zfunction-contracts` flag. If you would like to learn more about the development and implementation details of this feature and leave feedback please refer to [the RFC](https://model-checking.github.io/kani/rfc/rfcs/0009-function-contracts.html).
+
+## Introduction
+
+Verification is a costly art. If you are an active user of Kani you know that the runtime of a proof does not scale linearly with the amount of code under verification and you probably have a number of functions and ideas for harnesses you’d like Kani to verify, but it just takes prohibitively long.
+
+We [previously](https://model-checking.github.io/kani-verifier-blog/2023/02/28/kani-internship-projects-2022-stubbing.html) blogged about stubbing, a technique where a function that performs a costly computation, for instance trough recursion or a long running loop, could be replaced with a function that is cheaper to execute in the verifier. This replacement approximates the behavior of the function it replaces and/or uses techniques only available in a verifier, like non-deterministic values (`kani::any`). By successively replacing costly functions with stubs Kani can check ever larger proofs but it also comes with a downside.
```

Comment:
I would change the definition of stub, i.e., the sentence "a technique where...". Stubbing is more than that though. It can also be used to model IO, system calls, and other things that contracts cannot.

Once you define it, you mention that one use case is to replace costly computation.